src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -239,8 +312,48 @@ std::unordered_set< int > computeMPINeighborRanks( std::vector< vtkBoundingBox >
       neighbors.insert( assignments[i] );
     }
   }
+}
+
+/**
+ * @brief Redistribute the mesh among the available MPI ranks
+ * @details this method will also generate global ids for points and cells in the VTK Mesh
+ * @param[in] loadedMesh the mesh that was loaded on one or several MPI ranks
+ * @param[in,out] mpiNeighbors the set of neighboring MPI ranks, will be updated
+ */
+vtkSmartPointer< vtkUnstructuredGrid >
+redistributeMesh( vtkUnstructuredGrid & loadedMesh,
+                  bool const useGraphPartitioning,
+                  std::set< int > & mpiNeighbors )
+{
+  GEOSX_MARK_FUNCTION;
+
+  // Generate global IDs for vertices and cells
+  vtkNew< vtkGenerateGlobalIds > generator;
+  generator->SetInputDataObject( &loadedMesh );
+  generator->Update();
+
+  vtkIdType const minCellsOnAnyRank = MpiWrapper::min( loadedMesh.GetNumberOfCells() );
+  if( minCellsOnAnyRank > 0 )
+  {
+    // We read a properly pre-partitioned mesh, so there should be nothing to do
+    return vtkUnstructuredGrid::SafeDownCast( generator->GetOutputDataObject( 0 ) );
+  }
```

Comment:
Does this criterion for skipping the redistribution seem reasonable to everyone? The idea is that if every rank already has at least some cells, then we've likely loaded an already properly partitioned `.pvtu` dataset.